app.js

```diff
@@ -338,8 +337,21 @@ function findCompilers() {
         });
     }
 
+    function getTypeFromBin(exe) {
```

Comment:
That makes it harder to extend :) Next cab off the rank will be the D compiler. So made it an if else statement.